### PR TITLE
[WIP] Fixes #22917 - add fixtures to foreman tests

### PR DESCRIPTION
**Pull request by jlsherrill**
This uses the new foreman fixture plugin api to add katello fixtures
along side foreman ones.  In addition it restructures katello fixtures
to not require us to explicitly set the class for each one, as well
as moves them around so they can be auto loaded

### CODE REVIEW

**Comment by jlsherrill** [comment]
Requires https://github.com/theforeman/foreman/pull/5495

File: test/katello_test_helper.rb [hunk 1]
```diff
@@ -72,27 +71,20 @@ module FixtureTestCase
...
 
     Setting::Content.load_defaults
 
+    FIXTURES = load_fixtures(ActiveRecord::Base)
     @@admin = ::User.unscoped.find(FIXTURES['users']['admin']['id'])
     User.current = @@admin
   end
+
+  def katello_environments(name)
+    katello_k_t_environments(name)
```
Comment (akofink) [inline test/katello_test_helper.rb]: Where is this method defined?

File: test/fixtures/models/katello_pool_products.yml [hunk 2]
```diff
@@ -1,11 +0,0 @@
-basic_fedora:
-    pool_id: <%= ActiveRecord::FixtureSet.identify(:pool_one) %>
-    product_id: <%= ActiveRecord::FixtureSet.identify(:fedora) %>
-
-basic_redhat:
-    pool_id: <%= ActiveRecord::FixtureSet.identify(:pool_one) %>
-    product_id: <%= ActiveRecord::FixtureSet.identify(:redhat) %>
-
-other_sub_fedora:
```
Comment (akofink) [inline test/fixtures/models/katello_pool_products.yml]: So... these aren't used anywhere? Is that right?

**Comment by jlsherrill** [comment]
Closing, as i've figured out a simpler way to solve this :)

**Pull request closed without merging**